core/src/test/java/org/ehcache/EhcacheEventsTestBase.java

```diff
@@ -61,7 +62,8 @@
   protected Ehcache<String, String> getEhcache(final CacheLoaderWriter<String, String> cacheLoaderWriter, String name) {
     ExecutorService orderedExecutor = Executors.newSingleThreadExecutor();
     ExecutorService unorderedExecutor = Executors.newCachedThreadPool();
-    cacheEventNotificationService = new CacheEventNotificationServiceImpl<String, String>(orderedExecutor, unorderedExecutor, store);
+    TestTimeSource timeSource = new TestTimeSource();
```

Comment:
Why do you introduce a `TestTimeSource` here? Is it because the `SystemTimeSource` is not in scope as it lives in `impl` module?
Especially as the `TestTimeSource` provides the advanceTime method which is not used.
Two options:

* Move `SystemTimeSource from `impl` to `core`
* Use an anonymous inner class that wraps `System.currentTimeMillis()`